### PR TITLE
UI improvements

### PR DESCRIPTION
**Pull request by OlafPettersson**
I would like to proprose four small UI Improvements with this PR.

1. When on the map window, bring FleetWindows to the top of the Window Stack, even when the same fleet has be re-selected. At the moment, FleetWindow will stay hidden behind other windows, when clicking on the same fleet again (after interacting with other windows, e.g. the Pedia, which might have overlapped the fleet window).
2. When selecting a Planet, always bring the SidePanel - when it is visible - to the top of the Window Stack. At the moment the SidePanel will stay behind other windows, e.g. the Pedia. This makes overlapping windows on the screen tedious.

3. Add commands "bring to top/bring to bottom" to the Ship Design Window. 

4. When adding all Ship Designs from the saved folder, retain their relative position. At the moment, the Designs are added in reverse order to the available ship designs.

### CODE REVIEW

File: UI/DesignWnd.cpp [hunk 1]
```diff
@@ -3071,6 +3073,18 @@ void CompletedDesignsListBox::BaseRightClicked(GG::ListBox::iterator it, const G
         }
     };
 
+    auto movetotop_design_action = [&design, this]() {
+        auto saved_design = *design;
```
Comment (geoffthemedio) [inline UI/DesignWnd.cpp]: why the temporary instead of just using `->`?

File: UI/DesignWnd.cpp [hunk 2]
```diff
@@ -3142,11 +3161,24 @@ void SavedDesignsListBox::BaseRightClicked(GG::ListBox::iterator it, const GG::P
...
+        GetSavedDesignsManager().MoveBefore(design->UUID(), boost::uuids::uuid{{0}});
+        Populate();
+    };
+
     // add all saved designs
     auto add_all_saved_designs_action = [&manager, &empire_id]() {
         DebugLogger() << "BasesListBox::BaseRightClicked AddAllSavedDesignsToDisplayedDesigns";
-        for (const auto& uuid : manager.OrderedDesignUUIDs())
-            AddSavedDesignToDisplayedDesigns(uuid, empire_id);
+        // add the items to the end of the existing list, int correct order
```
Comment (geoffthemedio) [inline UI/DesignWnd.cpp]: int -> in ?

File: UI/DesignWnd.cpp [hunk 3]
```diff
@@ -3164,6 +3196,9 @@ void SavedDesignsListBox::BaseRightClicked(GG::ListBox::iterator it, const GG::P
     popup->AddMenuItem(GG::MenuItem(true)); // separator
     popup->AddMenuItem(GG::MenuItem(UserString("DESIGN_WND_DELETE_SAVED"),          false, false, delete_saved_design_action));
     popup->AddMenuItem(GG::MenuItem(true)); // separator
+    popup->AddMenuItem(GG::MenuItem(UserString("MOVE_UP_QUEUE_ITEM"), false, false, movetotop_design_action));
```
Comment (geoffthemedio) [inline UI/DesignWnd.cpp]: indent `false` to match surrounding lines

File: UI/MapWnd.cpp [hunk 4]
```diff
@@ -4570,14 +4565,16 @@ void MapWnd::SelectFleet(std::shared_ptr<Fleet> fleet) {
...
     // make sure selected fleet's FleetWnd is active
     manager.SetActiveFleetWnd(fleet_wnd);
     GG::GUI::GetGUI()->MoveUp(fleet_wnd);
     PushWndStack(fleet_wnd);
 
-
-
+    // if indicated fleet is already the only selected fleet in the active FleetWnd, 
+    // nothing to do.
+    if (m_selected_fleet_ids.size() == 1 && m_selected_fleet_ids.count(fleet->ID())){
```
Comment (geoffthemedio) [inline UI/MapWnd.cpp]: remove the `{` and `}`

**Comment by geoffthemedio** [comment]
The lists of designs on the design screen are not a queue, so shouldn't be called one.

Otherwise OK.

**Comment by OlafPettersson** [comment]
I made the coding style changes and introduced separate translations for moving items in a list. I also found that two translation changes had somehow slipped in from another branch of mine and reverted these with the last commit.

**Comment by geoffthemedio** [comment]
The design screen finished designs tab still says move to queue top/bottom.

**Comment by geoffthemedio** [comment]
@OlafPettersson will you address the finished designs "queue" naming issue?

**Pull request merged**